lib/assembly/build.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Assembly.Build do
+  @moduledoc """
+  Each process represents an active build watching for inventory changes and updating its status accordingly
+  """
+
+  use GenServer
+
+  alias Assembly.{Builder, Components}
+  alias Bottle.Assembly.V1.Build
+
+  def start_link(%Build{} = build) do
+    GenServer.start_link(__MODULE__, build)
+  end
+
+  @impl true
+  def init(%Build{} = build) do
+    {:ok, build}
+  end
+
+  @impl true
+  def handle_cast(:determine_status, %Build{build_components: components, status: current_status} = build) do
+    readyable? =
+      Enum.all?(components, fn %{id: component_id, quantity: quantity_needed} ->
+        {:ok, quantity} = Components.quantity_available(component_id)
+        not is_nil(quantity) and quantity >= quantity_needed
+      end)
+
+    build = %Build{build | status: build_status(readyable?)}
+
+    if (current_status == :BUILD_STATUS_READY) ^^^ readyable? do
```

Comment:
```suggestion
    if readyable? and (current_status != :BUILD_STATUS_READY) do
```

Reply:
The XOR here is important. There's 2 scenarios we want to notify people that builds have had a status change:

1. Build status changing from READY to INCOMPLETE
2. Build status changing from INCOMPLETE to READY

So

If `readyable?` = `true` and `status` = `:BUILD_STATUS_READY`, no update.
If `readyable?` = `true` and `status` != `:BUILD_STATUS_READY`, update.
If `readyable?` = `false` and `status` = `:BUILD_STATUS_READY`, update.
If `readyable?` = `false` and `status` != `:BUILD_STATUS_READY`, no update.

Reply:
Ah. I didn't realize that was an XOR. The github highlighting also messed up 😓